crates/uv-distribution/src/workspace.rs

```diff
@@ -824,6 +808,52 @@ fn is_excluded_from_workspace(
     Ok(false)
 }
 
+/// An iterator over a [`Pattern`] in a workspace.
+///
+/// Iterates from the workspace root, returning paths that match the pattern.
```

Comment:
It looks like this iterator _only_ returns directory paths? If so, I think that'd be good to document here.